nodejs/awsx/ecs/taskDefinition.ts

```diff
@@ -51,7 +51,7 @@ export abstract class TaskDefinition extends pulumi.ComponentResource {
 
     constructor(type: string, name: string,
                 isFargate: boolean, args: TaskDefinitionArgs,
-                opts?: pulumi.ComponentResourceOptions) {
+                opts: pulumi.ComponentResourceOptions) {
```

Comment:
it's an abstract base, and it helps ensure a pattern of requirin the derived type to remember to pass this in.  it's useful to prevent accidental omission (which has bit us in the past).